src/err/mod.rs

```diff
@@ -554,7 +567,7 @@ impl PyErr {
     where
         T: PyTypeInfo,
     {
-        self.is_instance(py, T::type_object_bound(py).as_gil_ref())
+        self.is_instance_bound(py, T::type_object_bound(py).as_any().clone())
```

Comment:
Once the input is changed to `&Bound` you can get away with `Deref` coercion to any:

```suggestion
        self.is_instance_bound(py, &T::type_object_bound(py))
```